src/imap.rs

```diff
@@ -92,6 +92,10 @@ pub struct Imap {
     interrupt: Option<stop_token::StopSource>,
     should_reconnect: bool,
     login_failed_once: bool,
+
+    /// True if CAPABILITY command was run successfully once and config.can_* contain correct
+    /// values.
+    capabilities_determined: bool,
```

Comment:
Because `connect()` is used during autoconfiguration, and running a `CAPABILITY` command is a waste of time for a connection that will be closed immediately. So I do it on demand, during `prepare()`.